### DISCUSSION TITLE
Readme Suggestions

### DISCUSSION THREAD

**Discussion by macqueenism** [graphql-hive/graphql-eslint#1517] [Ideas]
Hi, I've just migrated over to graphql-eslint from the eslint-plugin-graphql and found a couple of gotchas that might be worth adding to the README for future users. 

The first thing that caught me out was that the whole eslint config needs to be within the `overrides` section. I think I saw a comment about that in one of the issues but it could possibly do with being outlined as a warning for folks doing a migration. 

The second gotcha for me was that I don't use `graphql-config` in my project so it seems like including the `"skipGraphQLConfig": true` in parser options is a required step in this case. The wording of [that section of the docs](https://the-guild.dev/graphql/eslint/docs/getting-started/parser-options#skipgraphqlconfig) doesn't suggest it's required so more of a warning here would have helped me. 

Let me know if you think these would be of any use to other people. 

Thanks for all the hard work on this project!

**Comment by dimaMachina** [comment]
> The first thing that caught me out was that the whole eslint config needs to be within the overrides section. I think I saw a comment about that in one of the issues but it could possibly do with being outlined as a warning for folks doing a migration.

it's just as per ESLint work, you should configure parser/processor/rules to run on specific files, it's identical in other non-js plugins - `json`, `md`/`mdx`, `yml`/`yaml`, `vue`, `svelte` etc.

> The second gotcha for me was that I don't use graphql-config in my project so it seems like including the "skipGraphQLConfig": true in parser options is a required step in this case. The wording of [that section of the docs](https://the-guild.dev/graphql/eslint/docs/getting-started/parser-options#skipgraphqlconfig) doesn't suggest it's required so more of a warning here would have helped me.

GraphQL-ESLint firstly searches for the presence of a physical config, after for `parserOptions.schema/operations` setting, setting  `"skipGraphQLConfig": true` just disables the search for a physical config, so I'll suggest setting it up if you describe your schema/document pointers in eslint config.

Feel free to contribute to improving docs, I will be appreciated by it!

> Thanks for all the hard work on this project!

Thanks for using it!
